src/KnownRelaysManager.ts

```diff
@@ -0,0 +1,292 @@
+    [];
+
+  public allRelayers: IRelayHub.RelayManagerDataStruct[] = [];
+
+  constructor(
+    contractInteractor: ContractInteractor,
+    config: EnvelopingConfig,
+    relayFilter?: RelayFilter,
+    scoreCalculator?: AsyncScoreCalculator
+  ) {
+    this.config = config;
+    this.relayFilter =
+      relayFilter ??
+      function () {
+        return true;
+      };
+    this.scoreCalculator = scoreCalculator ?? defaultRelayScore;
+    this.contractInteractor = contractInteractor;
+    this.relayLookupWindowParts = this.config.relayLookupWindowParts;
+  }
+
+  async refresh(): Promise<void> {
+    this._refreshFailures();
+    log.debug('KnownRelaysManager - Refresh failures done');
+    const recentlyActiveRelayManagers =
+      await this._fetchRecentlyActiveRelayManagers();
+    log.debug(
+      'KnownRelaysManager - Fetched recently active Relay Managers done'
+    );
+
+    this.preferredRelayers = this.config.preferredRelays.map((relayUrl) => {
+      return {
+        url: relayUrl,
+      };
+    });
+
+    this.allRelayers = await this.getRelayDataForManagers(
+      recentlyActiveRelayManagers
+    );
+
+    log.debug('KnownRelaysManager - Get relay info for Managers done');
+    log.debug('KnownRelaysManager - Refresh done');
+  }
+
+  async getRelayDataForManagers(
+    relayManagers: Set<string>
+  ): Promise<IRelayHub.RelayManagerDataStruct[]> {
+    if (relayManagers.size === 0) {
+      return [];
+    }
+    const activeRelays = await this.contractInteractor.getActiveRelayInfo(
+      relayManagers
+    );
+    // As 'topics' are used as 'filter', having an empty set results in querying all register events.
+
+    return activeRelays.filter(this.relayFilter);
+  }
+
+  splitRange(
+    fromBlock: number,
+    toBlock: number,
+    splits: number
+  ): Array<{ fromBlock: number; toBlock: number }> {
+    const totalBlocks = toBlock - fromBlock + 1;
+    const splitSize = Math.ceil(totalBlocks / splits);
+
+    const ret: Array<{ fromBlock: number; toBlock: number }> = [];
+    let b;
+    for (b = fromBlock; b < toBlock; b += splitSize) {
+      ret.push({
+        fromBlock: b,
+        toBlock: Math.min(toBlock, b + splitSize - 1),
+      });
+    }
+
+    return ret;
+  }
+
+  // return events from hub. split requested range into "window parts", to avoid
+  // fetching too many events at once.
+  getPastEventsForHub(
+    fromBlock: number,
+    toBlock: number
+  ): Promise<TypedEvent[][]> {
+    const rangeParts = this.splitRange(
+      fromBlock,
+      toBlock,
+      this.relayLookupWindowParts
+    );
+
+    const pastEventsPromises = rangeParts.map(({ fromBlock, toBlock }) => {
+      return this.contractInteractor.getPastEventsForHub({
+        fromBlock,
+        toBlock,
+      });
+    });
+
+    return Promise.all(pastEventsPromises).then(
+      (pastEventArray: TypedEvent[][][]) => {
+        return pastEventArray.flat();
+      }
+    );
```

Comment:
I don't think we're covering the exact same functionality we had before. By reading the previous code, it looks like there was a sort of retrial mechanism that we're missing now. I know that we aren't going to use this component, but I'd keep the same logic, just not to forget about it.

Reply:
Restored